debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/StreamingWal2JsonMessageDecoder.java

```diff
@@ -0,0 +1,170 @@
+ *
+ * <p>There are four different chunks that can arrive from the decoder.
+ * <b>Beginning of message</b></br>
+ * <pre>
+ * {
+ *   "xid": 563,
+ *   "timestamp": "2018-03-20 10:58:43.396355+01",
+ *   "change": [
+ * </pre>
+ *
+ * <b>First change</b>
+ * <pre>
+ *      {
+ *          "kind": "insert",
+ *          "schema": "public",
+ *          "table": "numeric_decimal_table",
+ *          "columnnames": ["pk", "d", "dzs", "dvs", "d_nn", "n", "nzs", "nvs", "d_int", "dzs_int", "dvs_int", "n_int", "nzs_int", "nvs_int", "d_nan", "dzs_nan", "dvs_nan", "n_nan", "nzs_nan", "nvs_nan"],
+ *          "columntypes": ["integer", "numeric(3,2)", "numeric(4,0)", "numeric", "numeric(3,2)", "numeric(6,4)", "numeric(4,0)", "numeric", "numeric(3,2)", "numeric(4,0)", "numeric", "numeric(6,4)", "numeric(4,0)", "numeric", "numeric(3,2)", "numeric(4,0)", "numeric", "numeric(6,4)", "numeric(4,0)", "numeric"],
+ *          "columnoptionals": [false, true, true, true, false, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true],
+ *          "columnvalues": [1, 1.10, 10, 10.1111, 3.30, 22.2200, 22, 22.2222, 1.00, 10, 10, 22.0000, 22, 22, null, null, null, null, null, null]
+ *      }
+ * </pre>
+ *
+ * <b>Further changes</b>
+ * <pre>
+ *      ,{
+ *          "kind": "insert",
+ *          "schema": "public",
+ *          "table": "numeric_decimal_table",
+ *          "columnnames": ["pk", "d", "dzs", "dvs", "d_nn", "n", "nzs", "nvs", "d_int", "dzs_int", "dvs_int", "n_int", "nzs_int", "nvs_int", "d_nan", "dzs_nan", "dvs_nan", "n_nan", "nzs_nan", "nvs_nan"],
+ *          "columntypes": ["integer", "numeric(3,2)", "numeric(4,0)", "numeric", "numeric(3,2)", "numeric(6,4)", "numeric(4,0)", "numeric", "numeric(3,2)", "numeric(4,0)", "numeric", "numeric(6,4)", "numeric(4,0)", "numeric", "numeric(3,2)", "numeric(4,0)", "numeric", "numeric(6,4)", "numeric(4,0)", "numeric"],
+ *          "columnoptionals": [false, true, true, true, false, true, true, true, true, true, true, true, true, true, true, true, true, true, true, true],
+ *          "columnvalues": [1, 1.10, 10, 10.1111, 3.30, 22.2200, 22, 22.2222, 1.00, 10, 10, 22.0000, 22, 22, null, null, null, null, null, null]
+ *      }
+ * </pre>
+ *
+ * <b>End of message</b>
+ * <pre>
+ *      ]
+ * }
+ * </pre>
+ * </p>
+ *
+ * <p>
+ * For parsing purposes it is necessary to add or remove a fragment of JSON to make a well-formatted JSON out of it.
+ * The last message is just dropped.
+ * </p>
+ * @author Jiri Pechanec
+ *
+ */
+public class StreamingWal2JsonMessageDecoder implements MessageDecoder {
+
+    private static final  Logger LOGGER = LoggerFactory.getLogger(StreamingWal2JsonMessageDecoder.class);
+
+    private final DateTimeFormat dateTime = DateTimeFormat.get();
+    private boolean containsMetadata = false;
+    private boolean messageInProgress = false;
+    private String bufferedContent;
```

Comment:
I'm still trying to figure why we need the buffer here at all. Can we simply consume each chunk as it arrives? You seem to do some delayed processing just if the next message arrives. Why is this?

Reply:
@gunnarmorling 
Check this - https://github.com/debezium/debezium/pull/480/files#diff-77ee0f8ecffcd58e3f29e08fd55dc9d6R148

We need to know which of the changes is the last one in the batch. It is easy to recognize if you read the message as the whole. In case of chunked message you need to wait till you get the end of message chunk. Hence the delayed sending - I can decide if this change is the last one or not onyl if I receive the next one.